File: ios/CameraManager.m
#import <Foundation/Foundation.h>
#import <React/RCTViewManager.h>
#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(CameraManager, RCTViewManager)
RCT_EXPORT_VIEW_PROPERTY(flash, BOOL)

RCT_EXTERN_METHOD(capture:(nonnull NSNumber *)node)
@end

File: ios/CameraManager.swift
import Foundation
import UIKit

@objc(CameraManager)
class CameraManager: RCTViewManager {
  override func view() -> UIView! {
    return CameraView()
  }
  
  override static func requiresMainQueueSetup() -> Bool {
    return true
  }
  
  @objc
  func capture(_ node: NSNumber) {
    let component = getCameraView(tag: node)
    component.capture()
  }
  
  func getCameraView(tag: NSNumber) -> CameraView {
    return bridge.uiManager.view(forReactTag: tag) as! CameraView
  }
}
